torchvision/prototype/transforms/functional/_augment.py

```diff
@@ -35,6 +35,8 @@ def erase(
     inplace: bool = False,
 ) -> Union[features.ImageTypeJIT, features.VideoTypeJIT]:
     if isinstance(inpt, torch.Tensor):
+        if not torch.jit.is_scripting() and isinstance(inpt, (features.Image, features.Video)):
+            inpt = inpt.as_subclass(torch.Tensor)
```

Comment:
Unfortunately we are forced to double estimate the `if` statement. JIT has a special treatment for handling `is_scripting()` if inlined. To avoid estimating this twice, let's switch to the idiom used everywhere else. Aka:
- if pure tensor => kernel call, no wrapping
- if Image or Video => kernel call, wrapping
- else => PIL